src/parser.h

```diff
@@ -267,16 +348,118 @@ typedef struct {
         char const *labels[TRANSACTION_PROMPT_BATCH_SIZE + 1]; // For NULL at end
         prompt_entry_t entries[TRANSACTION_PROMPT_BATCH_SIZE];
     } prompt;
+    uint32_t raw_type_id;
     enum transaction_type_id_t type_id;
+    bool is_p_chain;
+    bool is_x_chain;
+    bool is_c_chain;
     bool swap_output;
     uint64_t last_output_amount;
     network_id_t network_id;
     uint64_t sum_of_inputs;
     uint64_t sum_of_outputs;
     uint64_t staking_weight;
     uint64_t staked;
+
+
 } parser_meta_state_t;
 
+// EVM stuff below this line
+
+enum assetCall_state_t {
+    ASSETCALL_ADDRESS,
+    ASSETCALL_ASSETID,
+    ASSETCALL_AMOUNT,
+    ASSETCALL_DATA,
+    ASSETCALL_DONE,
+};
+
+struct EVM_ABI_state { };
+
+struct EVM_assetCall_state {
+  enum assetCall_state_t state;
+  uint64_t data_length;
+    union {
+        struct Id32_state id32_state;
+        struct uint256_t_state uint256_state;
+        struct uint32_t_state selector_state;
+        struct {
+            struct Address_state address_state;
+            parser_input_meta_state_t chunk;
+            // union EVM_endpoint_states endpoint_state;
+        };
+    };
+};
+
+union EVM_endpoint_states {
+    struct EVM_ABI_state abi_state;
+    struct EVM_assetCall_state assetCall_state;
+};
+
+struct struct_evm_parser_meta_state_t;
+typedef struct struct_evm_parser_meta_state_t evm_parser_meta_state_t;
+
+typedef enum parse_rv (*known_destination_init)(union EVM_endpoint_states *const state, uint64_t length);
+typedef enum parse_rv (*known_destination_parser)(union EVM_endpoint_states *const state, parser_input_meta_state_t *const input, evm_parser_meta_state_t *const meta);
+
+struct known_destination {
+  uint8_t to[20];
+  known_destination_init init_value;
+  known_destination_parser handle_value;
```

Comment:
What are these two for?

Reply:
Handling the value field when we're doing a known contract. If we aren't currently showing the avax value of an assetCall, we should put functions here to demand that it be 0. If we're showing the transfer part anyways then we don't strictly need them.